### ISSUE TITLE
Fuzz testing RPC

### ISSUE THREAD

**Issue by ilblackdragon** [near/nearcore#2422]
Fuzzer to test RPC connections, by sending:

*   [ ] Invalid methods
*   [ ] Correct method with invalid formatted data
*   [ ] Correct method with validly formatted random data

**Comment by bowenwang1996** [comment]
@frol please take a look. I believe this is quite important for phase 0.

**Comment by frol** [comment]
While it is definitely important, I think it is not a blocker for phase 0. Though, I will take a closer look into the closed connection issue reported in https://github.com/nearprotocol/nearcore/issues/2402#issuecomment-612321881. #2074 is another item required to be implemented before even trying to fuzz-test the API.

**Comment by ilblackdragon** [comment]
Confirming not a blocker for Phase 0. Moved to Phase 1.

**Comment by nearmax** [comment]
If it is in the IceBox then it cannot be in Phase 1. We need to decide to either move it from Phase 1 or from the Ice Box. Also, is it something we can ask @khorolets to do?

**Comment by khorolets** [comment]
Explain me please what to do and I am ready to help. 

@frol am I capable to handle it? :)

**Comment by frol** [comment]
We should resolve the blocking items first. Specifically, #2074 (also in Icebox at the moment).

If this is Phase 1 blocker, we should definitely postpone Network Indexer and work on these two items. /cc @kcole16

@khorolets don't ask me, ask Rust compiler :)

I will on-board @khorolets for this task and fix #2074 myself if we prioritize these as Phase 1.

**Comment by bowenwang1996** [comment]
@frol sounds like a solid plan.

**Comment by frol** [comment]
The first iteration is done, and we identified a few issues (see the issues that mention this issue) which block us from running fuzz further.

**Comment by ilblackdragon** [comment]
Pushing to the next release as the main work is done and it's not blocked by P2 issues to be fixed.

**Comment by bowenwang1996** [comment]
@frol what is the latest on this?
